### PR TITLE
Configuration switch for search service integration

### PR DESCRIPTION
**Pull request by SharmaAjay19**

### CODE REVIEW

File: src/Diagnostics.RuntimeHost/Controllers/DiagnosticControllerBase.cs [hunk 1]
```diff
@@ -571,24 +574,38 @@ private async Task<IEnumerable<DiagnosticApiResponse>> ListDetectorsInternal(Run
...
+                            {
+                                p.EntryPointDefinitionAttribute.Score = 0;
+                            }
+                        });
+                        allDetectors = allDetectors.Where(x => x.EntryPointDefinitionAttribute.Score > 0).ToList();
                     }
                     else
                     {
-                        p.EntryPointDefinitionAttribute.Score = 0;
+                        throw new Exception("Search API returned null");
```
Comment (ShekharGupta1988) [inline src/Diagnostics.RuntimeHost/Controllers/DiagnosticControllerBase.cs]: are we throwing a 500 if search service returned no detectors?

Comment (ShekharGupta1988) [reply src/Diagnostics.RuntimeHost/Controllers/DiagnosticControllerBase.cs]: if yes, then we should return [] with a 200 Response. 
If this is some error condition, a more verbose error message will help in production.

File: src/Diagnostics.RuntimeHost/Services/SearchService/SearchService.cs [hunk 2]
```diff
@@ -30,54 +34,97 @@ public class SearchService : ISearchService
...
+                isEnabled = Convert.ToBoolean(Registry.GetValue(RegistryConstants.SearchAPIRegistryPath, RegistryConstants.SearchAPIEnabledKey, string.Empty));
+            }
+            else
+            {
+                isEnabled = Convert.ToBoolean(_config[$"SearchAPI:{RegistryConstants.SearchAPIEnabledKey}"]);
+            }
         }
 
-        public Task<HttpResponseMessage> Get(HttpRequestMessage request)
+        public async Task<HttpResponseMessage> Get(HttpRequestMessage request)
```
Comment (ShekharGupta1988) [inline src/Diagnostics.RuntimeHost/Services/SearchService/SearchService.cs]: out of curiosity, why this change?  looks like you dont need to await on any statement in this method

File: src/Diagnostics.RuntimeHost/Services/SearchService/SearchService.cs [hunk 3]
```diff
@@ -30,54 +34,97 @@ public class SearchService : ISearchService
...
+                parameters.Add("requestId", requestId ?? string.Empty);
+                HttpRequestMessage request = new HttpRequestMessage(HttpMethod.Post, QueryDetectorsUrl);
+                request.Content = new StringContent(JsonConvert.SerializeObject(parameters), Encoding.UTF8, "application/json");
+                return await Get(request);
+            }
+            return null;
         }
 
-        public Task<HttpResponseMessage> SearchUtterances(string requestId, string query, string[] detectorUtterances, Dictionary<string, string> parameters)
+        public async Task<HttpResponseMessage> SearchUtterances(string requestId, string query, string[] detectorUtterances, Dictionary<string, string> parameters)
```
Comment (ShekharGupta1988) [inline src/Diagnostics.RuntimeHost/Services/SearchService/SearchService.cs]: same here

File: src/Diagnostics.RuntimeHost/Controllers/DiagnosticControllerBase.cs [hunk 4]
```diff
@@ -571,24 +574,38 @@ private async Task<IEnumerable<DiagnosticApiResponse>> ListDetectorsInternal(Run
...
                     }
-                });
-                allDetectors = allDetectors.Where(x => x.EntryPointDefinitionAttribute.Score > 0).ToList();
+                }
+                else
+                {
+                    throw new Exception("Search service is disabled.");
```
Comment (dagrooms52) [inline src/Diagnostics.RuntimeHost/Controllers/DiagnosticControllerBase.cs]: This would still break the local environment if we disable search service.

I have a branch with changes to disable search service - https://github.com/Azure/Azure-AppServices-Diagnostics/blob/c91f0cc20707bfa82d6d53d1d9ef8aae8899f634/src/Diagnostics.RuntimeHost/Controllers/DiagnosticControllerBase.cs
